src/surface.lisp

```diff
@@ -309,6 +309,31 @@ Otherwise, return the copy of the image data along with the pointer."
   (with-cairo-object (surface pointer)
 	(cairo_image_surface_get_stride pointer)))
 
+(defmacro with-surface ((surface &optional surface-name) &body body)
+  (let ((var-name (or surface-name '*surface*)))
+    `(let ((,var-name ,surface))
+       (unwind-protect
+	    (progn ,@body)
+       
+	 (progn (surface-finish ,var-name)
+		(destroy ,var-name))))))
+
```

Comment:
I feel like this should optionally not destroy the surface (maybe use `&key` not `&optional` for further extension).  Also, there's no `(defvar *surface*)` afaict, which there probably should be, even if it's not really used elsewhere else (yet!).